src/main/java/games/strategy/util/Version.java

```diff
@@ -57,7 +57,13 @@ public Version(final String version) {
         m_point = 0;
       }
       if (tokens.hasMoreTokens()) {
-        m_micro = Integer.parseInt(tokens.nextToken());
+        int micro;
+        try {
+          micro = Integer.parseInt(tokens.nextToken());
+        } catch (NumberFormatException e) {
+          micro = 99999;
```

Comment:
Should `micro` be set to `99999` for _any_ non-integer token or only for `"dev"`?  We probably want consistency between this parsing logic and the formatting logic in `toString()` and `toStringFull()`.

Should the constants `99999` and `"dev"` be extracted to fields since they are used in multiple locations?

We probably need a few unit tests here to cover the new branches.

Reply:
I'm not sure why we would get a number format? If there are no tokens, then we ran off the end, otherwise would we not expect a valid number token?

Reply:
Would it maybe make sense to just hard code the micro version always?

Reply:
Seriously dudes, do these things really matter?

I don't see the importance of preventing the run time error in the dev versions (although I guess that could be annoying).

> I'm not sure why we would get a number format? If there are no tokens, then we ran off the end, otherwise would we not expect a valid number token?

tokens are strings.

Reply:
> Seriously dudes, do these things really matter?

Yes, why do things the complicated way when you do not have to?

> I don't see the importance of preventing the run time error in the dev versions

I don't see how you ever would get a runtime error. Are you proposing we by hand attach '.dev' to the game engine version? If there are only 3 number tokens, and we parse them all, I don't see how we would get a number format problem. Meanwhile, simply hardcoding the micro version seems way simpler than adding a dummy magic value that we then parse and handle in a very special way.

Reply:
What I am proposing is to reduce complexity!

I could explain about the runtime error but it's a thing of the past.